phylanx/ir/node_data.hpp

```diff
@@ -218,6 +242,14 @@ namespace phylanx { namespace ir
             increment_move_construction_count();
         }
 
+        template <typename U,
+            typename U1 =
+                typename std::enable_if<!std::is_same<T, U>::value>::type>
+        node_data(node_data<U> const& d)
+          : data_(init_data_from_type(d))
+        {
+        }
+
```

Comment:
Shouldn't be something similar be added for `node_data& operator=(node_data<U> const&)`?